waspc/cli/src/Wasp/Cli/Interactive.hs

```diff
@@ -0,0 +1,86 @@
+module Wasp.Cli.Interactive
+  ( askForInput,
+    askToChoose,
+    askForRequiredInput,
+  )
+where
+
+import Data.Foldable (find)
+import Data.List (intercalate)
+import qualified Data.Text as T
+import System.IO (hFlush, stdout)
+import Text.Read (readMaybe)
+import qualified Wasp.Util.Terminal as Term
+
+askForRequiredInput :: String -> IO String
+askForRequiredInput question = do
+  ensureNotNull . askForInput $ question
+
+askToChoose :: Show a => String -> [a] -> IO a
+askToChoose question options = case options of
+  [] -> error "Cannot ask to choose from empty list of options."
+  [singleOption] -> return singleOption
+  _multipleOptionsCase -> askUserToChoose
+  where
+    askUserToChoose = do
+      putStrLn $ Term.applyStyles [Term.Bold] question
+      putStrLn optionsToChooseFrom
+      answer <- prompt
+
+      handleOptionIndex answer >>= \case
+        Just option -> return option
+        Nothing -> handleOptionName answer
+
+    handleOptionIndex answer = do
+      let index = parseIndexOrDefault answer
+
+      case index of
+        Just i -> do
+          if i `elem` indexes
+            then return $ Just $ options !! (i - 1)
+            else Just <$> invalidOptionAction
+        Nothing -> return Nothing
+      where
+        -- The default option is the first one
+        parseIndexOrDefault input = if null input then Just 1 else readMaybe input :: Maybe Int
+
+    handleOptionName answer = do
+      let option = findElemByString answer
+      case option of
+        Just o -> return o
+        Nothing -> invalidOptionAction
+
+    invalidOptionAction = do
+      putStrLn $ Term.applyStyles [Term.Red] "Invalid selection, write the name or the index of the option."
+      askToChoose question options
+
+    findElemByString option = find (\o -> show o == option) options
+
+    optionsToChooseFrom = intercalate "\n" $ prependIndexToOption options
+    prependIndexToOption = zipWith (\i o -> showIndexWithStyle i ++ show o ++ noteForDefault i) indexes
```

Comment:
Probably should be named `prependIndexesToOptions` (I added "s" at the end).
It returns String though, that is not very clear.